l10n_eu_product_adr_delivery_carrier/models/delivery_carrier.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class DeliveryCarrier(models.Model):
+
+    _inherit = "delivery.carrier"
+
+    adr_limited_amount_ids = fields.Many2many(
+        "limited.amount",
+        string="Warns according to ADR limited amount",
+        help="Raise warning when following limited amount are defined on "
+        "any product in the document.",
+    )
+
+    def _prepare_adr_warning(self, warnings, checked_object):
+        """Prepare warning dictionnary to be displayed according to warnings
+
+        :param warnings: Dict of {item: warning_record} where:
+          - item is either a sale.order.line or stock.move object
+          - warning_record is an object that defines field name
+        :param checked_object: String being either "order_line" or "move" that
+          must match items in warnings parameter
+        :return: Warning dictionnary
+        """
+        self.ensure_one()
+        single_object_str = ""
+        checked_object_str = ""
+        if checked_object not in ("move", "order_line"):
+            raise UserError(
+                _(
+                    "checked_object parameter %s used to call "
+                    "_prepare_dangerous_goods_warning is invalid."
+                )
+                % checked_object
+            )
+        elif checked_object == "move":
```

Comment:
I'd drop this part and pass the string as an argument. For plural issue, just use "move(s)".